src/countries/countries.service.ts

```diff
@@ -1,4 +1,46 @@
 import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Country } from './country.entity';
+import { DeleteResult, Repository, UpdateResult } from 'typeorm';
+import { CountryInterface } from './country.interface';
 
 @Injectable()
-export class CountriesService {}
+export class CountriesService {
+  constructor(
+    @InjectRepository(Country)
+    private countryRepository: Repository<CountryInterface>,
+  ) {}
+
+  findAll(): Promise<CountryInterface[]> {
+    return this.countryRepository.find();
+  }
+
+  findOne(id: number): Promise<CountryInterface | null> {
+    return this.countryRepository.findOneById(id);
+  }
+
+  create(country: CountryInterface): Promise<CountryInterface> {
+    return this.countryRepository.save(this.countryRepository.create(country));
+  }
+
+  update(id: number, country: CountryInterface): Promise<UpdateResult> {
+    return this.countryRepository
+      .createQueryBuilder()
```

Comment:
Any specific reason behind choosing a Query Builder instead of the TypeORM update function?
`await this.countryRepository.update(id, countryDto);`